pubtools/pulplib/_impl/fake/client.py

```diff
@@ -13,6 +14,7 @@
 from .match import match_object
 
 Publish = namedtuple("Publish", ["repository", "tasks"])
+Upload = namedtuple("Upload", ["repo_id", "tasks"])
```

Comment:
I'm not quite sure if this API with Upload and upload_history is complete enough.  I see a couple of issues with it:

Since it doesn't give you any info about what content was uploaded, all you can really check is whether the count of uploads is what you expected per repo.  So, for instance, if you had a class which was generating content and uploading them to Pulp repos, you couldn't use the fake client to check anything about the content uploaded, whether they were uploaded with correct filenames, etc.  It's really quite limited what you can do here.  This isn't a deal breaker since it could potentially have more attributes added later.

The bigger issue is - what about other content types in the future?  Right now, you've got upload_history and we only support file uploads.  Later we will support other kinds of uploads.  I'm thinking that it's quite possible different types of uploads ought to have different attributes collected in the history, so it may be a mistake to add one API which claims to support all kinds of uploads.

Can you please consider what to do about these issues?  Also, it's always an option to leave out that new history API altogether now and let someone add it when they have a use-case for it.